src/modules/product-category/product-category.controller.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller, Get, Post, Body, Param, Patch, Delete, UseGuards, Query } from '@nestjs/common';
+import { ProductCategoryService } from './product-category.service';
+import { CreateProductCategoryDto } from './dto/create-product-category.dto';
+import { UpdateProductCategoryDto } from './dto/update-product-category.dto';
+import { ApiTags, ApiBody, ApiParam } from '@nestjs/swagger';
+import { skipAuth } from '../../helpers/skipAuth';
+@ApiTags('products/categories')
+@Controller('products/categories')
+export class ProductCategoryController {
+  constructor(private readonly categoryService: ProductCategoryService) {}
+
+  @Get()
+  async findAll(@Query('limit') limit?: number, @Query('offset') offset?: number) {
+    return {
+      status_code: 200,
+      categories: await this.categoryService.getAllCategories(limit, offset),
+    };
+  }
+
+  @Get(':id')
+  @ApiParam({ name: 'id', description: 'Category ID', example: 'uuid-of-category' })
+  async findOne(@Param('id') id: string) {
+    return {
+      status_code: 200,
+      category: await this.categoryService.getCategoryById(id),
+    };
+  }
+
+  @skipAuth()
```

Comment:
Is this route protected or public. If protected, you need to remove this decorator. 
The function of the skipAuth decorator is to bypass authentication

Reply:
it is protected. we needed to test the routes with no token from users